sdk/dotnet/PicovoiceTest/MainTest.cs

```diff
@@ -49,69 +49,199 @@ public static void ClassInitialize(TestContext testContext)
 
-            int framecount = (int)Math.Floor((float)(data.Count / _pv.FrameLength));
+            List<short> data = GetPcmFromFile(testAudioPath, _picovoice.SampleRate);
+
+            int framecount = (int)Math.Floor((float)(data.Count / _picovoice.FrameLength));
             var results = new List<int>();
             for (int i = 0; i < framecount; i++)
             {
-                int start = i * _pv.FrameLength;
-                int count = _pv.FrameLength;
+                int start = i * _picovoice.FrameLength;
+                int count = _picovoice.FrameLength;
                 List<short> frame = data.GetRange(start, count);
 
-                _pv.Process(frame.ToArray());
+                _picovoice.Process(frame.ToArray());
             }
 
             Assert.IsTrue(_isWakeWordDetected);
-            Assert.AreEqual(_inference.Intent, "orderBeverage");
-            Dictionary<string, string> expectedSlotValues = new Dictionary<string, string>()
+            Assert.AreEqual(_inference.Intent, expectedIntent);
+
+            Assert.IsTrue(_inference.Slots.All((keyValuePair) =>
+                                          expectedSlots.ContainsKey(keyValuePair.Key) &&
+                                          expectedSlots[keyValuePair.Key] == keyValuePair.Value));
+        }
+
+        [TestMethod]
+        public void TestTwice()
+        {
+            string language = "en";
+            _picovoice = Picovoice.Create(
+                ACCESS_KEY,
+                GetKeywordPath(language, "picovoice"),
+                _wakeWordCallback,
+                GetContextPath(language, "coffee_maker"),
+                _inferenceCallback);
+
+            string audioFileName = "picovoice-coffee.wav";
+            string expectedIntent = "orderBeverage";
+            Dictionary<string, string> expectedSlots = new Dictionary<string, string>()
             {
                 {"size", "large"},
                 {"beverage", "coffee"}
             };
-            Assert.IsTrue(_inference.Slots.All((keyValuePair) =>
-                                          expectedSlotValues.ContainsKey(keyValuePair.Key) &&
-                                          expectedSlotValues[keyValuePair.Key] == keyValuePair.Value));
+            RunTestCase(audioFileName, expectedIntent, expectedSlots);
+            TestInit();
+            RunTestCase(audioFileName, expectedIntent, expectedSlots);
         }
 
         [TestMethod]
-        public void TestProcessAgain()
+        public void TestTwiceDe()
         {
-            TestProcess();
+            string language = "de";
+            _picovoice = Picovoice.Create(
+                ACCESS_KEY,
+                GetKeywordPath(language, "heuschrecke"),
+                _wakeWordCallback,
+                GetContextPath(language, "beleuchtung"),
+                _inferenceCallback,
+                GetPorcupineModelPath(language),
+                0.5f,
+                GetRhinoModelPath(language));
```

Comment:
I don't believe you need the `0.5f` as that should be a default. You may have to use named arguments to trigger that though:
```csharp
porcupineModelPath: GetPorcupineModelPath(language),
rhinoModePath: GetRhinoModelPath(language)
```